modules/tensor_mechanics/include/materials/czm/CZMMaterialBase.h

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InterfaceMaterial.h"
+class CZMMaterialBase;
+template <>
+InputParameters validParams<CZMMaterialBase>();
+/**
+ * This is the base Material class for implementing a traction separation material model.
+ * The responsibility of this class is to rotate the displacement jump from global to local
+ * coordinate and rotate back traction and traction derivatives.
+ * The local coordinate system assumes the following order: opening,
+ * tangential1, tangential2. Note that tangential1, tangential2 are arbitrary and therefore the
+ * interface assumes an in-plane isotropic behavior. By subclassing computeTraction and
```

Comment:
Instead of 'subclassing', say  'overriding computeTraction and computeTractionDerivatives' in a derived class'

Reply:
updated the comment.
Also please check I did what you suggested earlier